domain/user/service/service.go

```diff
@@ -136,12 +137,28 @@ func ValidateUsername(name string) error {
 // - usererrors.AlreadyExists: If a user with the supplied name already exists.
 // - usererrors.UserCreatorNotFound: If a creator has been supplied for the user
 // and the creator does not exist.
-func (s *Service) AddUser(ctx context.Context, user user.User) error {
+func (s *Service) AddUser(ctx context.Context, user user.User, creatorUsername string) error {
```

Comment:
Should we accept a UUID for the creator user instead of a name?

Reply:
So, this question is worth the discussion.

The current position is "UUID should exist only on the state layer." Because it means nothing to all other components.

That is why the service layer provides interfaces only with names and a stale layer  - with UUIDs.

Reply:
UUID should absolutely not be confined to the state layer. Juju should evolve such that it is present for all entity types, including the DTOs.

The rationale being that if I create a machine client side, it can have a UUID and be uniquely identified within the system at inception, even before we have persisted it. This will become more valuable over time, particularly as usage scenarios become richer - think Terraform, MicroCloud, Sunbeam.

Reply:
got it,

I provided a commit that made all functions in the service layer use UUIDs instead of Names. 
Also, it drops domainUser as we do not need it anymore.